### PR TITLE
Created Solution for "CSES 1188 - Bit Inversions"

### PR DESCRIPTION
**Pull request by dutinmeow**

### CODE REVIEW

**Review by sathu-beep — CHANGES_REQUESTED** [review]
lmfao it failed vercel find out why

**Comment by dutinmeow** [comment]
The error message says that the internal solution to the problem is missing. I'm not sure as to if this is a normal problem for changing the JSON file and uploading a new solution or if my file name is just wrong.

File: solutions/cses-1188.mdx [hunk 1]
```diff
@@ -0,0 +1,127 @@
+---
+id: CSES-1188
```
Comment (bqi343) [inline solutions/cses-1188.mdx:2]: ^ should be `cses-1188`

**Review by bqi343 — CHANGES_REQUESTED** [review]

**Comment by bqi343** [comment]
> lmfao it failed vercel find out why

not helpful ...

**Comment by sathu-beep** [comment]
> > lmfao it failed vercel find out why
> 
> not helpful ...

😔 sorry next time I will try to be more helpful

File: solutions/cses-1188.mdx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+- The longest prefix of the same bit in the range $[l, r]$ (denote this as $P$)
+- The longest suffix of the same bit in the range $[l, r]$ (denote this as $S$)
+- The longest subarray of the same bit in the range $[l, r]$ (denote this as $A$)
+- The length of the subarray (equivalent to $r - l + 1$) (denote this as $L$)
+
+To merge left child $u$ and right child $v$ to make node $w$
+- If $u.P = u.L$, then $w.P = u.L + v.P$; otherwise, $w.P = u.P$
+- If $v.S = v.L$, then $w.S = v.S + u.S$; otherwise, $w.S = v.S$
+- $w.A = \max(u.A, v.A, u.S + v.P)$
+- $w.L = u.L + v.L$
```
Comment (bqi343) [inline solutions/cses-1188.mdx:18]: ^ why not use the same names as your code below?

File: solutions/cses-1188.mdx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+Clearly, a bit flip operation corresponds to a point update on the segment tree and a longest homogeneous subarray query corresponds to the $A$ value of
```
Comment (bqi343) [inline solutions/cses-1188.mdx:20]:
```suggestion
A bit flip operation corresponds to a point update on the segment tree and a longest homogeneous subarray query corresponds to the $A$ value of 
```

File: solutions/cses-1188.mdx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+source: CSES
+title: Bit Inversions
+author: Dustin Miao
+---
+
+We can use a Segment Tree over the given array of bits. In a node of the segment tree corresponding to range $[l, r]$, we will store three pieces of information:
+- The longest prefix of the same bit in the range $[l, r]$ (denote this as $P$)
+- The longest suffix of the same bit in the range $[l, r]$ (denote this as $S$)
+- The longest subarray of the same bit in the range $[l, r]$ (denote this as $A$)
+- The length of the subarray (equivalent to $r - l + 1$) (denote this as $L$)
```
Comment (bqi343) [inline solutions/cses-1188.mdx:12]: ^ this assumes that you are looking for the longest run of a fixed bit value (e.g. the longest run of 1s)

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/cses-1188.mdx [hunk 5]
```diff
@@ -5,21 +5,22 @@ title: Bit Inversions
...
+- The longest subarray of1s in the range $[l, r]$ (denote this as $A$)
 - The length of the subarray (equivalent to $r - l + 1$) (denote this as $L$)
 
-To merge left child $u$ and right child $v$ to make node $w$
-- If $u.P = u.L$, then $w.P = u.L + v.P$; otherwise, $w.P = u.P$
-- If $v.S = v.L$, then $w.S = v.S + u.S$; otherwise, $w.S = v.S$
-- $w.A = \max(u.A, v.A, u.S + v.P)$
-- $w.L = u.L + v.L$
+To merge left child $a$ and right child $b$ to make node $c$
+- If $a.P = u.L$, then $c.P = a.L + b.P$; otherwise, $c.P = a.P$
```
Comment (bqi343) [inline solutions/cses-1188.mdx:16]: still occurrences of `u` and `v`

File: solutions/cses-1188.mdx [hunk 6]
```diff
@@ -5,21 +5,22 @@ title: Bit Inversions
...
+- If $b.S = v.L$, then $c.S = b.S + a.S$; otherwise, $c.S = b.S$
```
Comment (bqi343) [inline solutions/cses-1188.mdx:17]:
```suggestion
- If $b.S = b.L$, then $c.S = b.L + a.S$; otherwise, $c.S = b.S$
```

so it's consistent w/ code below

**Review by bqi343 — CHANGES_REQUESTED** [review]

**Review by bqi343 — APPROVED** [review]

**Review by dongliu0426 — APPROVED** [review]
Can also be done with one segtree

**Pull request merged**